R/nitrogen.R

```diff
@@ -0,0 +1,107 @@
+#' Calculate the NLV
+#' 
+#' This function calculates the NLV (nitrogen producing capacity) for the soil
+#' 
+#' @param n.org (numeric) The organic nitrogen content of the soil in g  N / kg
+#' @param c.org (numeric) The organic carbon content of the soil in kg C / ha
+#' @param crop (numeric) The crop code (gewascode) from the BRP
+#' @param soiltype (character) The type of soil
+#' @param bulk_density (numeric) The bulk density of the soil in kg / m3
+#' @param cn_ratio (numeric) 
+#' @param grass.age (numeric) The age of the grass if present
+#' 
+#' @import data.table
+#' 
+#' @export
+calc_nlv <- function(n.org, c.org, crop, soiltype, bulk_density, cn_ratio, grass.age) {
+  
+  # Load in the datasets
+  crops.obic <- as.data.table(OBIC::crops.obic)
+  setkey(crops.obic, crop_code)
+  soils.obic <- as.data.table(OBIC::soils.obic)
+  setkey(soils.obic, soiltype)
+  
+  # Check input
+  arg.length <- max(length(n.org), length(c.org), length(crop), length(soiltype), length(bulk_density), length(grass.age))
+  checkmate::assert_numeric(n.org, lower = 0, upper = 100, any.missing = FALSE, len = arg.length)
```

Comment:
lower and lower is quite limited for N and C as the units are kg / ha...?

Reply:
Solved see 0abb80583f8c1350cdce37ffb031f6ade00ef648